docs/orchestrating-elastic-stack-applications/logstash.asciidoc

```diff
@@ -0,0 +1,569 @@
+
+NOTE: The `clusterName` value should be unique across namespaces.
+
+NOTE: The logstash ECK operator will create a user called `eck_logstash_user_role` when an `elasticsearchRef` is specified. This user has the following permissions:
+```
+  "cluster": ["monitor", "manage_ilm", "read_ilm", "manage_logstash_pipelines", "manage_index_templates", "cluster:admin/ingest/pipeline/get",],
+  "indices": [
+    {
+      "names": [ "logstash", "logstash-*", "ecs-logstash", "ecs-logstash-*", "logs-*", "metrics-*", "synthetics-*", "traces-*" ],
+      "privileges": ["manage", "write", "create_index", "read", "view_index_metadata"]
+    }
+
+```
+The permissions for this user <<{p}-users-and-roles,can be updated>> to include more indices if the elasticsearch plugin is expected to use indices other than the default. See <<{p}-logstash-configuration-examples, logstash configuration examples>> for a sample configuration that creates a user to write to a custom index.
+
+The following example demonstrates how to create a Logstash deployment that connects to
+different elasticsearch instances, one of which is in a separate namespace:
+
+[source,yaml,subs="attributes,+macros,callouts"]
+----
+apiVersion: logstash.k8s.elastic.co/v1alpha1
+kind: Logstash
+metadata:
+  name: quickstart
+spec:
+  version: {version}
+  count: 1
+  elasticsearchRefs:        <1>
+    - clusterName: prod-es  <2>
+      name: prod
+    - clusterName: qa-es    <3>
+      name: qa
+      namespace: qa
+  pipelines:
+    - pipeline.id: main
+      config.string: |
+        input {
+          beats {
+            port => 5044
+          }
+        }
+        output {
+          elasticsearch {   <4>
+            hosts => [ "${PROD_ES_ES_HOSTS}" ]
+            user => "${PROD_ES_ES_USER}"
+            password => "${PROD_ES_ES_PASSWORD}"
+            cacert => "${PROD_ES_ES_SSL_CERTIFICATE_AUTHORITY}"
+          }
+          elasticsearch {   <4>
+            hosts => [ "${QA_ES_ES_HOSTS}" ]
+            user => "${QA_ES_ES_USER}"
+            password => "${QA_ES_ES_PASSWORD}"
+            cacert => "${QA_ES_ES_SSL_CERTIFICATE_AUTHORITY}"
+          }
+        }
+
+----
+
+<1> Define Elasticsearch references in the CRD. This will create the appropriate secrets to store certificate details and the rest of the connection information, and create environment variables to allow them to be referred to in Logstash pipeline configurations.
+<2> This refers to an Elasticsearch cluster residing in the same namespace as the logstash instances
+<3> This refers to an Elasticsearch cluster residing in a different namespace to the logstash instance
+<4> Elasticsearch output definitions - use the environment variables created by the Logstash operator when specifying an `ElasticsearchRef`. Note the use of "normalized" versions of the `clusterName` in the environment variables used to populate the relevant fields
```

Comment:
```suggestion
<1> Define Elasticsearch references in the CRD. This will create the appropriate Secrets to store certificate details and the rest of the connection information, and create environment variables to allow them to be referred to in Logstash pipeline configurations.
<2> This refers to an Elasticsearch cluster residing in the same namespace as the Logstash instances.
<3> This refers to an Elasticsearch cluster residing in a different namespace to the Logstash instances.
<4> Elasticsearch output definitions - use the environment variables created by the Logstash operator when specifying an `ElasticsearchRef`. Note the use of "normalized" versions of the `clusterName` in the environment variables used to populate the relevant fields.
```